sdk/iot/core/inc/az_iot_core.h

```diff
@@ -120,7 +117,7 @@ AZ_NODISCARD int32_t az_iot_retry_calc_delay(
  * occurrence of (but not including the) `delimiter`, or the end of `source` if `delimiter` is not
  * found. If `source` is empty, AZ_SPAN_NULL is returned instead.
  */
-AZ_NODISCARD az_span az_span_token(az_span source, az_span delimiter, az_span* out_remainder);
+AZ_NODISCARD az_span _az_span_token(az_span source, az_span delimiter, az_span* out_remainder);
```

Comment:
Based on the method name this should be in az_span.  This also sounds like it might be a json parser api.  Is there a reason for not adding to az_span or using the json parser?

Reply:
That's right and the reason I moved it to internal - it should definitely not be a public API within the IoT headers.
There was some discussion a while back (@JeffreyRichter @ewertons have more context) and it was decided to remain outside of the span API - not sure why.

I do think the token API is useful: 1. we used it and liked it, 2. C's StdLib provides `strtok`

Reply:
I will defer this to offline discussions and a later PR.

Reply:
It should not be public and I think it was moved into core (maybe its still an outstanding PR?).
Maybe @ahsonkhan or @vhvb1989 know?

Reply:
It hasn't moved to core yet. Final decision on semantics of the API was pending, which is why we decided to leave it in iot/common for now (btw, we probably want to rename iot/core to iot/common to avoid confusion).

We can (and should) move it to core in the near future (post preview?).